x-pack/legacy/plugins/ml/public/data_frame_analytics/pages/analytics_management/components/create_analytics_flyout/create_analytics_flyout.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiTitle,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { CreateAnalyticsFormProps } from '../../hooks/use_create_analytics_form';
+
+export const CreateAnalyticsFlyout: FC<CreateAnalyticsFormProps> = ({
+  actions,
+  children,
+  state,
+}) => {
+  const { closeModal, createAnalyticsJob, startAnalyticsJob } = actions;
+  const { isJobCreated, isJobStarted, isModalButtonDisabled, isValid } = state;
+
+  return (
+    <EuiFlyout size="s" onClose={closeModal}>
+      <EuiFlyoutHeader>
+        <EuiTitle>
+          <h2 data-test-subj="mlDataFrameAnalyticsFlyoutHeaderTitle">
+            {i18n.translate('xpack.ml.dataframe.analytics.create.modalHeaderTitle', {
+              defaultMessage: 'Create analytics job',
+            })}
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>{children}</EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        {(!isJobCreated || !isJobStarted) && (
+          <EuiButtonEmpty onClick={closeModal}>
+            {i18n.translate('xpack.ml.dataframe.analytics.create.modalCancelButton', {
+              defaultMessage: 'Cancel',
```

Comment:
The label of this button should switch to 'Close' if the user opts just to create the job. After the job is created, but not yet started, the label 'Cancel' might imply that hitting the button will also cancel creation of the job.

Reply:
Good one! Added to my follow-up list